GrayBMP.cs

```diff
@@ -169,3 +184,26 @@ void Fatal (string message)
    #endregion
 }
 #endregion
+
+#region Struct Point2D
+
+record struct Point2D (double X, double Y) {
+   /// <summary> Returns the projected point at given distance at given angle</summary>
+   public static Point2D Project (Point2D p, double d, double theta) => new (p.X + (d * Cos (theta)), p.Y + (d * Sin (theta)));
+   /// <summary> Returns vertices of polygon which encloses the given two points</summary>
+   public static Point2D[] GetPolygonVertices (Point2D startPt, Point2D endPt, double offset) {
+      double angle = Atan2 (endPt.Y - startPt.Y, endPt.X - startPt.X);
+      var pts = new Point2D[8];
+      int count = 0;
+      double startAngle = PI / 2,
+             endAngle = 3 * PI / 2,
+             increment = PI / 3;
+      for (double i = startAngle; i <= endAngle; i += increment) {

```

Comment:
1. Don't use `i` as the name of a double variable. i typically is used only to refer to an _integer_ variable. 
2. This loop will not necessarily work correctly. As the termination condition, you are checking that i <= endangle. However, they are both double-precision variables and using ==, <= or >= on a double is never a good idea. Since you know exactly how many iterations the loop should run, use an integer to control that.